libraries/node-core-library/src/LockFile.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as path from 'path';
+import * as child_process from 'child_process';
+
+export function getProcessStartTime(pid: string): string | undefined {
+  let args: string[];
+  if (process.platform === 'darwin') {
+    args = [`-p ${pid}`, '-f'];
+  } else if (process.platform === 'linux') {
+    args = ['-p', pid.toString(), '-f'];
+  } else {
+    throw new Error(`Unsupported system: ${process.platform}`);
+  }
+
+  const psResult: string = child_process.spawnSync('ps', args).stdout.toString();
```

Comment:
>ps [](start = 52, length = 2)

Is it necessary to rely on PATH here?

Reply:
It should be in `/usr/bin`

---
In reply to: [165795924](https://github.com/Microsoft/web-build-tools/pull/517#discussion_r165795924) [](ancestors = 165795924)